### ISSUE TITLE
Mouse position is offset depending on node position and zoom level in GraphEdit

### ISSUE THREAD

**Issue by charlesmlamb** [godotengine/godot#61449]
### Godot version

4.0 dev

### System information

Windows 10 . .

### Issue description

It's difficult to capture with an image, problem is if I try and select a port . .

![image](https://user-images.githubusercontent.com/83387915/170604625-eb07d10a-9507-4c2c-b182-203a48709b0b.png)

Mostly, it captures the port right over the one I'm trying to 'click' on . . It happens at all zoom levels, and is particularly annoying when working at a high zoom level, where atm I constantly click the wrong port, it's for visual shaders, but, also visual script, not sure what is happening, it looks like active area is vertically 'wrong' . .

### Steps to reproduce

Make a visual shader, visual script, etc . . Try to select a port, esp. when zoomed out, it tends to get it wrong, the mouse is poorly aligned . . .

Beyond that, Godot 4 is, 1.4 - 1.8 x as good . . . #thx . .

### Minimal reproduction project

Like above, mouse doesn't work, constantly selects wrong port, so on . . .

**Comment by viksl** [comment]
I can confirm this. I've added a video to showcase this.

It seems the the dots (connections) get misaligned at certain zoom level and based on position of the node on the screen in the editor.

In the video I initially show how if I try to grab X coordinate depending on where the node is put I either can grab the point or I drag the entire node although everytime I click on the dot.

Mid of the video you can see I added a new node to showcase the same issue to demonstrate it's not just one node problem.

At about 1:21 I showcase the same zoom as ath the beginning, same node, same connection point but I panned the screen a bit and suddenly I can grab the point just fine so it's seems to be related not just to the zoom level but also position.

At the last part of the video I showcase that you can grab the point but you need to move your cursor quite a bit above the point so the visual representation isn't where the editor actually detects it.

Video link: https://youtu.be/pqlrPDD3aPw

This has been around for ages ;).

**Comment by KoBeWi** [comment]
Is this still reproducible in GraphEdit in 4.0?

**Comment by viksl** [comment]
> Is this still reproducible in GraphEdit in 4.0?

I've been trying to reproduce it now for a while past hour and haven't been able to, not saying it can't happen but nothing from what was messed up before in 3.x seems to be wrong now in 4.0.
I tested it in the visual shader window, previously in 3.x I also tested it in visual shader.

Maybe someone else can try reproduce this in 4.0 because I can't no matter zoom or position it always seems to work now at least as far as I can tell, also the area to grab the point got larger, whoever finally made that thank you!
